packages/@n8n/ai-workflow-builder.ee/evaluations/chains/evaluators/connections-evaluator.ts

```diff
@@ -52,13 +52,67 @@ Your task is to verify that every connection follows n8n's sourcing rules, suppo
 ## Patterns to Validate
 
 ### Retrieval-Augmented Generation (RAG)
-- Data source or metadata prep (HTTP Request, Set, etc.) → Vector Store **via main** when mode is insert/upsert
-- Token Splitter → Document Loader [ai_textSplitter]
-- Document Loader → Vector Store [ai_document]
-- Embeddings → Vector Store [ai_embedding]
+
+**CRITICAL: Document Loader is a CAPABILITY-ONLY sub-node. It NEVER receives main data flow.**
+
+The Document Loader node in n8n:
+- Has NO main input - it cannot and should not receive data via main connections
+- ONLY connects via ai_document TO a Vector Store (Document Loader is the SOURCE, Vector Store is the TARGET)
+- Reads data from the workflow context (binary files, JSON) based on its configuration
+- Is a capability provider, not a data processor in the main flow
+
+**CORRECT RAG Pipeline Pattern:**
+\`\`\`
+Data Source (Extract From File, HTTP Request, etc.)
+       │
+       │ [main]
+       ▼
+Vector Store (insert mode) ◄──[ai_document]── Document Loader ◄──[ai_textSplitter]── Text Splitter
+       ▲
+       │
+       └──[ai_embedding]── Embeddings
+\`\`\`
+
+**How it works:**
+1. Data source connects to Vector Store via **main** - this triggers the insert operation
+2. Document Loader connects TO Vector Store via **ai_document** - provides document processing capability
+3. Text Splitter connects TO Document Loader via **ai_textSplitter** - provides chunking capability
+4. Embeddings connects TO Vector Store via **ai_embedding** - provides vectorization capability
+
+**THE FOLLOWING ARE ALL CORRECT - NEVER FLAG AS VIOLATIONS:**
+- Document Loader has NO main connections (no main input, no main output) - THIS IS CORRECT BY DESIGN
+- Document Loader connects TO Vector Store via ai_document - THIS IS THE ONLY WAY TO USE DOCUMENT LOADER
+- Extract From File/PDF/CSV connects directly to Vector Store via main - THIS IS CORRECT
+- Data flows: Extract → Vector Store (main) while Document Loader → Vector Store (ai_document) - BOTH ARE NEEDED, BOTH ARE CORRECT
+- Document Loader appears "disconnected" from the main workflow path - THIS IS CORRECT, IT ONLY USES ai_document
+
+**INVALID VIOLATION EXAMPLES - DO NOT OUTPUT THESE:**
```

Comment:
The double negative is confusing and might be confusing for the agent as the context grows. Could we frame it in more positive clearer terms. 
i.e. Here are valid examples
Here are violations